sfaira/models/embedding/output_layers.py

```diff
@@ -91,7 +91,116 @@ def call(self, inputs, **kwargs):
         var_clip = tf.clip_by_value(var, -bound, bound, "decoder_clip")
 
-        exp_mean = tf.exp(mean_clip + sf)
-        exp_var = tf.exp(var_clip)
+        invlinker_mean = tf.exp(mean_clip + sf)
+        invlinker_var = tf.exp(var_clip)
 
-        return [exp_mean, exp_var]
+        return [invlinker_mean, invlinker_var]
+
+
+class GaussianOutput(tf.keras.layers.Layer):
+    """
+    Gaussian output layer.
+
+    Size factor only makes sense if logged and data is positive and logged.
+    """
+
+    def __init__(
+            self,
+            original_dim=None,
+            name='neg_bin_output',
+            **kwargs
+    ):
+
+        super().__init__(name=name, **kwargs)
+
+        self.means = tf.keras.layers.Dense(original_dim, activation='linear')
+        self.var = tf.keras.layers.Dense(original_dim, activation='linear')
+
+    def call(self, inputs, **kwargs):
+        activation, sf = inputs
+        mean, var = self.means(activation), self.var(activation)
+
+        # clip to log of largest values supported by log operation
+        bound = 60.
+        mean_clip = tf.clip_by_value(mean, tf.exp(-bound), tf.exp(bound), "decoder_clip")
+        var_clip = tf.clip_by_value(var, -bound, bound, "decoder_clip")
+
+        invlinker_mean = mean_clip + sf
+        invlinker_var = tf.exp(var_clip)
+
+        return [invlinker_mean, invlinker_var]
+
+
+class GaussianSharedStdOutput(tf.keras.layers.Layer):
+    """
+    Gaussian output layer with a single standard deviation estimate per features.
+
+    Size factor only makes sense if logged and data is positive and logged.
+    """
+
+    def __init__(
+            self,
+            original_dim=None,
+            name='neg_bin_shared_disp_output',
+            **kwargs
+    ):
+
+        super().__init__(name=name, **kwargs)
+
+        self.means = tf.keras.layers.Dense(original_dim, activation='linear')
+        self.var = self.add_weight(
+            "var_bias",
+            shape=[1, original_dim]
+        )
+
+    def call(self, inputs, **kwargs):
+        activation, sf = inputs
+        mean = self.means(activation)
+        var = self.var
+        var = tf.broadcast_to(var, tf.shape(mean))
+
+        # clip to log of largest values supported by log operation
+        bound = 60.
+        mean_clip = tf.clip_by_value(mean, tf.exp(-bound), tf.exp(bound), "decoder_clip")
+        var_clip = tf.clip_by_value(var, -bound, bound, "decoder_clip")
+
+        invlinker_mean = mean_clip + sf
+        invlinker_var = tf.exp(var_clip)
+
+        return [invlinker_mean, invlinker_var]
+
+
+class GaussianConstStdOutput(tf.keras.layers.Layer):
+    """
+    Gaussian output layer with standard deviation set as constant (=1).
+
+    Size factor only makes sense if logged and data is positive and logged.
+    """
+
+    def __init__(
+            self,
+            original_dim=None,
+            name='neg_bin_const_disp_output',
```

Comment:
@davidsebfischer did you on purpose not change the name of the output layer here?